src/unity/toolkits/one_shot_object_detection/one_shot_object_detector.cpp

```diff
@@ -0,0 +1,171 @@
+#include <unity/toolkits/one_shot_object_detection/one_shot_object_detector.hpp>
+
+#include <unity/toolkits/object_detection/object_detector.hpp>
+
+// TODO: Clean up imports.
+#include <algorithm>
+#include <array>
+#include <cmath>
+#include <cstdio>
+#include <iostream>
+#include <limits>
+#include <numeric>
+#include <queue>
+#include <utility>
+#include <vector>
+#include <random>
+
+#include <logger/assertions.hpp>
+#include <logger/logger.hpp>
+#include <random/random.hpp>
+
+#include <boost/gil/gil_all.hpp>
+#include <boost/gil/extension/io/jpeg.hpp>
+#include <boost/gil/extension/numeric/sampler.hpp>
+#include <boost/gil/extension/numeric/resample.hpp>
+#include <boost/gil/utilities.hpp>
+
+#include <image/numeric_extension/perspective_projection.hpp>
+
+#include <Eigen/Core>
+#include <Eigen/Dense>
+
+using turi::coreml::MLModelWrapper;
+using namespace Eigen;
+using namespace boost::gil;
+
+namespace turi {
+namespace one_shot_object_detection {
+
+/* A ParameterSweep class to randomly generate different samples of parameters
+ * that can later be used to compute the transformation matrix necessary to 
+ * create image projections.
+ */
+class ParameterSweep {
```

Comment:
This class doesn't read like something that sweeps parameters or that is a sweep of parameters. It reads like something that samples parameters. But the parameters aren't clearly defined?

Reply:
Done.